src/parser/jobs/nin/modules/Ninjutsu.js

```diff
@@ -35,19 +36,21 @@ export default class Ninjutsu extends Module {
 		this.addEventHook('complete', this._onComplete)
 	}
 
-	_onDotonCast() {
+	_onDotonCast(event, fabricated) {
 		this._finishDotonWindow()
 
 		this._dotonCasts.current = {
 			tcj: this.combatants.selected.hasStatus(STATUSES.TEN_CHI_JIN.id),
 			ticks: [],
+			prepull: fabricated && (event.timestamp - PREPULL_DOTON_BUFFER) <= this.parser.fight.start_time,
+			// Flag Dotons with no cast event and a first damage tick at or before 3s as prepull so we don't ding on them
 		}
 	}
 
 	_onDotonDamage(event) {
 		// If there are no casts at all, use the damage event to fabricate one
 		if (!this._dotonCasts.current) {
-			this._onDotonCast()
+			this._onDotonCast(event, true)
```

Comment:
On the one hand, I feel like fixing the data files so that PrecastStatus synthesizes the Doton cast would be a better approach (since Doton applies a status to the NIN that falls off when the AOE ends and so PCS can synth the cast).  On the other hand, I suppose this is the least cost change for prepull Doton tracking, since we're already synthing that first Doton cast anyway...